openbox/config.h

```diff
@@ -152,6 +152,8 @@ extern gchar *config_title_layout;
 extern gboolean config_animate_iconify;
 /*! Size of icons in focus switching dialogs */
 extern guint config_theme_window_list_icon_size;
+/*! Set a default icon for windows that lack one */
+extern gboolean config_apply_default_icon;
```

Comment:
The new option will need to be added to a few more places:
* The default rc.xml: https://github.com/Mikachu/openbox/blob/53737a844d2bba4bb5892737b05a01d6d06fe84b/data/rc.xml
* The rc.xsd: https://github.com/Mikachu/openbox/blob/53737a844d2bba4bb5892737b05a01d6d06fe84b/data/rc.xsd (it's a type `ob:bool` value)
* And the rc-mouse-focus.xml example config: https://github.com/Mikachu/openbox/blob/53737a844d2bba4bb5892737b05a01d6d06fe84b/doc/rc-mouse-focus.xml